app/jobs/publish_feed_job.rb

```diff
@@ -75,11 +75,23 @@ def save_file(podcast, feed, options = {})
     opts[:bucket] = s3_bucket
     opts[:key] = feed.path
     @put_object = s3_client.put_object(opts)
-    Episode.where(id: rss.episodes).where(first_rss_published_at: nil).update_all(first_rss_published_at: DateTime.now) if feed.default?
+
+    update_first_publish_episodes(podcast.first_publish_episodes) if feed.default?
```

Comment:
Hrm... I don't think this `first_publish_episodes` method on the podcast is going to work.  It's not filtering for episodes we just wrote to the RSS (`rss.episodes`).  So right now, this would immediately set `first_rss_published_at` on any draft/scheduled episodes  as well.

I think you'll need to keep that same `.where(id: rss.episodes)` around.  Either inline in this method, or move `def first_publish_episodes` from the Podcast to here.